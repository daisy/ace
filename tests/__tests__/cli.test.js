'use strict';

const ace = require('../runAceCLI');
const pkg = require('../../package');

const path = require('path');


test('test no input', () => {
  const { stdout, stderr, status } = ace([]);
  expect(status).toBe(1);
  expect(stderr).toBe('');
  expect(stdout).toMatchSnapshot();
});

test('test help', () => {
  const { stdout, stderr, status } = ace(['-h']);
  expect(status).toBe(0);
  expect(stderr).toBe('');
  expect(stdout).toMatchSnapshot();
});

test('test version -v', () => {
  const { stdout, stderr, status } = ace(['-v']);
  expect(status).toBe(0);
  expect(stderr).toBe('');
  expect(stdout.trim()).toBe(pkg.version);
});

test('test version --version', () => {
  const { stdout, stderr, status } = ace(['--version']);
  expect(status).toBe(0);
  expect(stderr).toBe('');
  expect(stdout.trim()).toBe(pkg.version);
});

test('test unexisting input', () => {
  const { stdout, stderr, status } = ace(['unexisting.epub']);
  expect(status).toBe(1);
  expect(stdout.trim()).toBe('');
  expect(stderr).toMatchSnapshot();
});

test('test existing output', () => {
  const { stdout, stderr, status } = ace(['-o', 'report', 'foo'], {
    cwd: path.resolve(__dirname, '../data'),
  });
  expect(status).toBe(1);
  expect(stderr).toBe('');
  expect(stdout).toMatchSnapshot();
});
