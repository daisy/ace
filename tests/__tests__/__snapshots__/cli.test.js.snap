// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test existing output 1`] = `
"Output directory is not empty.

Running Ace would override the following files or directories:

  - report/ace.json
  - report/report.html
  - report/js

Use option --force to override.

"
`;

exports[`test help 1`] = `
"
  Ace by DAISY, an Accessibility Checker for EPUB

  Usage: ace [options] <input>

  Options:

    -h, --help             output usage information
    -v, --version          output the version number

    -o, --outdir  <path>   save final reports to the specified directory
    -t, --tempdir <path>   specify a custom directory to store the temporary reports
    -f, --force            override any existing output file or directory
        --subdir           output reports to a sub-directory named after the input EPUB

    -V, --verbose          display verbose output
    -s, --silent           do not display any output

  Examples
    $ ace -o out ~/Documents/book.epub
"
`;

exports[`test no input 1`] = `
"Input required

  Ace by DAISY, an Accessibility Checker for EPUB

  Usage: ace [options] <input>

  Options:

    -h, --help             output usage information
    -v, --version          output the version number

    -o, --outdir  <path>   save final reports to the specified directory
    -t, --tempdir <path>   specify a custom directory to store the temporary reports
    -f, --force            override any existing output file or directory
        --subdir           output reports to a sub-directory named after the input EPUB

    -V, --verbose          display verbose output
    -s, --silent           do not display any output

  Examples
    $ ace -o out ~/Documents/book.epub
"
`;

exports[`test unexisting input 1`] = `
"[31merror[39m:   Couldnâ€™t find EPUB file 'unexisting.epub'
"
`;
